File: renders/facilities/Entrance.cpp
/* NIM/Nama  : 13515AAA / AAA AAA   */
/* File      : Entrance.cpp         */

#include <iostream>
#include "Entrance.h"
using namespace std;

/***** ENTRANCE *****/
Entrance::Entrance()  : id('i'), color(WHITE)
{}

Entrance::Entrance(const Entrance&)  : id('i'), color(WHITE)
{}

Entrance::~Entrance()
{}

Entrance& Entrance::operator=(const Entrance&)
{}

Entrance* Entrance::clone() const {
  return new Entrance(*this);
}

File: Entrance.java
/**
 * @author NIM / Nama : 13515087 / Audry Nyonata
 *  Nama file : Entrance.java
 */

import java.util.*;
import java.lang.*;

public class Entrance extends Road{
   /**
   * Constructor
   */
  public Entrance(){
    super();
  }
  /**
   * Mengembalikan karakter untuk ditampilkan ke layar
   * @return id bertipe char
   */
  public char Render(){
    return 'i';
  }
  /**
   * Mengembalikan warna untuk ditampilkan ke layar
   * @return color warna renderable
   */
  public String RenderWithColor(){
    return ANSI_WHITE + Render() + ANSI_RESET;
  }
}
